src/hardware/dac.rs

```diff
@@ -75,14 +75,37 @@ static mut DAC_BUF: [[SampleBuffer; 2]; 2] = [[[0; SAMPLE_BUFFER_SIZE]; 2]; 2];
 #[derive(Copy, Clone)]
 pub struct DacCode(pub u16);
 
-#[allow(clippy::from_over_into)]
-impl Into<f32> for DacCode {
-    fn into(self) -> f32 {
+impl From<f32> for DacCode {
+    fn from(voltage: f32) -> DacCode {
```

Comment:
The problem is that we're setting this from the telemetry `settings` interface. We currently don't have a way to propagate an invalid setting back to the user, which is why this is implemented in a saturating form. We could optionally just log the settings failure instead (and maybe add an MQTT topic for logs?). Open to suggestions here as well. If we just logged the failure, it would be confusing because the MQTT interface "accepts" the setting and responds with an OK before the value is validated.

Reply:
Then we still need to fail here and let the `settings_update` discard the error. Then later we can do something better with it (propagate the settings update success to the miniconf Ok MQTT response). I'd rather discard the update silently now than clip silently.